addons/mail/static/src/models/composer/composer.js

```diff
@@ -270,6 +272,136 @@ function factory(dependencies) {
             await this.env.bus.trigger('do-action', { action, options });
         }
 
+        parseBody() {
+            const escapedAndCompactContent = escapeAndCompactTextContent(this.textInputContent);
+            let body = escapedAndCompactContent.replace(/&nbsp;/g, ' ').trim();
+            body = this._generateMentionsLinks(body);
+            body = parseAndTransform(body, addLink);
+            body = this._generateEmojisOnHtml(body);
+            return body;
+        }
+
+        // this function will send the message inside the chatbox
+        // and queue it for sending.
+        queueMessage() {
+            const message = {
+                id: this.env.models['mail.message'].getNextTemporaryId(),
+                attachment_ids: this.attachments.map(attachment => attachment.id),
+                body: this.parseBody(),
+                channel_ids: this.mentionedChannels.map(channel => channel.id),
+                message_type: 'comment',
+                partner_ids: this.recipients.map(partner => partner.id),
+                subject: (this.subjectContent) ? this.subjectContent : undefined,
+                author: [['link', this.env.messaging.currentPartner]],
+                originThread: [['insert', {
+                    id: this.thread.id,
+                    model: this.thread.model,
+                }]],
+            };
+            this.update({
+                queue: [['insert', message]]
+            });
+            this._reset();
+            if (!this.queueProcess) {
+                this.postQueueMessage();
+            }
+        }
+
+        async postQueueMessage() {
+            this.update({
+                queueProcess: true,
+            });
+            const message = this.queue[0];
+            await this.postMessageQueued(message);
+            this.update({
+                queue: [['unlink', message]]
+            });
+            if (this.queue.length) {
+                this.postQueueMessage();
+            } else {
+                this.update({
+                    queueProcess: false,
+                });
+            }
+        }
+
+        async postMessageQueued(message) {
```

Comment:
I don't understand the name/use difference between `postQueueMessage` and `postMessageQueued`. Feels like an arbitrary split, probably need a single method? Or find better name (and add docstrings).